ckan_datapackage_tools/converter.py

```diff
@@ -177,82 +174,169 @@ def _parse_notes(dataset_dict):
-    if dataset_dict.get('license_url'):
-        license['url'] = dataset_dict['license_url']
+    keywords = [tag['name'] for tag in dataset_dict.get('tags', [])]
 
-    if license:
-        result['license'] = license
+    if keywords:
+        result['keywords'] = keywords
 
     return result
 
 
-def _parse_author_and_source(dataset_dict):
+def _parse_extras(dataset_dict):
     result = {}
-    source = {}
-
-    if dataset_dict.get('author'):
-        source['name'] = dataset_dict['author']
-    if dataset_dict.get('author_email'):
-        source['email'] = dataset_dict['author_email']
-    if dataset_dict.get('url'):
-        source['web'] = dataset_dict['url']
+    extras = [[extra['key'], extra['value']] for extra
+              in dataset_dict.get('extras', [])]
+    for extra in extras:
+        try:
+            extra[1] = json.loads(extra[1])
+        except (ValueError, TypeError):
+            pass
+    sources = _parse_organization_as_primary_source(dataset_dict)
+    licenses = _parse_primary_license(dataset_dict)
+    contributors = _parse_primary_contributors(dataset_dict)
+    if extras:
+        extras_dict = dict(extras)
+        result.update(_parse_profile(extras_dict))
+        _parse_extra_sources(extras_dict, sources)
+        _parse_extra_licenses(extras_dict, licenses)
+        _parse_extra_contributors(extras_dict, contributors)
+        if extras_dict:
+            result['extras'] = extras_dict
+    if sources:
+        result['sources'] = sources
+    if licenses:
+        result['licenses'] = licenses
+    if contributors:
+        result['contributors'] = contributors
+    return result
 
-    if source:
-        result['sources'] = [source]
 
+def _parse_profile(dataset_dict):
+    result = {}
+    profile = dataset_dict.pop('profile', '')
+    if profile:
+        result['profile'] = profile
     return result
 
 
-def _parse_maintainer(dataset_dict):
-    result = {}
-    author = {}
+def _parse_organization_as_primary_source(dataset_dict):
+    all_sources = []
+    source = {}
+    organization = dataset_dict.get('organization')
+    if organization.get('is_organization', False):
+        source['title'] = organization.get('name', organization.get('title'))
+    if source.get('title'):
+        if dataset_dict.get('url'):
+            source['path'] = dataset_dict['url']
+        all_sources.append(source)
+    return all_sources
 
-    if dataset_dict.get('maintainer'):
-        author['name'] = dataset_dict['maintainer']
-    if dataset_dict.get('maintainer_email'):
-        author['email'] = dataset_dict['maintainer_email']
 
-    if author:
-        result['author'] = author
+def _parse_extra_sources(extras_dict, sources):
+    for source in extras_dict.pop('sources', []):
+        parsed = _parse_extra_source(source)
+        if parsed and parsed.get('title'):
+            sources.append(parsed)
 
-    return result
 
+def _parse_extra_source(extra_source):
```

Comment:
I would expect any incoming `sources` property to be stored as-is, ie in the DP format, so it should be just a matter of dumping whatever is stored in the `sources` extra into the DP's `sources` property. eg sources in extras won't have `name` and `url`, they'll have whatever is on the DP spec

Reply:
Yes, I've since just done the same (sorry haven't pushed up next changes yet, but will asap).

Reply:
Oh, maybe I misunderstood the issue. My point is that why the stored `sources` in the CKAN extra should have a `name` and `url` property if these are not part of the spec but maybe what you are trying to do is support old versions of the spec? If that't the case it makes perfect sense, apologies.

Reply:
Hi @amercader 
Yes that's correct - sorry I'll make this more clear next time